tests/dojo_tests.py

```diff
@@ -6,17 +6,35 @@ def test_create_room_successfully(self):
         ndojo = Dojo()
         initial_room_count = len(ndojo.all_rooms)
         self.assertEqual(initial_room_count,0)
-        blue_office = ndojo.create_room("Blue","office")
+        blue_office = ndojo.create_room(["Blue"],"office")
         self.assertTrue(blue_office)
         new_room_count = len(ndojo.all_rooms)
         self.assertEqual(new_room_count - initial_room_count,1)
 
     def test_create_multiple_rooms_successfully(self):
         ndojo = Dojo()
         initial_room_count = len(ndojo.all_rooms)
-        ndojo.create_room(["Blue", "Green", "Purple"], "office")
+        log = ndojo.create_room(["Blue", "Green", "Purple"], "office")
+        self.assertTrue(log)
         new_room_count = len(ndojo.all_rooms)
         self.assertEqual(new_room_count - initial_room_count,3)
         ndojo.create_room(["Yellow", "Brown", "Black"], "livingspace")
         latest_room_count = len(ndojo.all_rooms)
         self.assertEqual(latest_room_count - new_room_count,3)
+    def test_check_invalid_input(self):
+        ndojo = Dojo()
+        log = ndojo.create_room([],"  ")
+        self.assertEqual(log,"Cannot create rooms with empty room name and/or empty room type")
+        log = ndojo.create_room([" "],"office")
+        self.assertEqual(log,"\nThe office at index 0 cannot be created due to empty name.")
+        log = ndojo.create_room(["Green", " ", "Black"],"livingspace")
+        self.assertEqual(log,"\nThe livingspace at index 1 cannot be created due to empty name.")
+        log = ndojo.create_room(["Green", "Black"],"piper")
+        self.assertEqual(log,"\nCannot creae room(s), invalid room type enterred")
```

Comment:
I like the way you have tested multiple scenarios. However, since each test scenario test for different errors, I think it would be better to split up this test case into multiple test cases, each testing a single thing. It makes for better unit tests.